mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -534,8 +577,15 @@ class Word2VecModel private[spark] (
     // Need not divide with the norm of the given vector since it is constant.
     val cosVec = cosineVec.map(_.toDouble)
     var ind = 0
+    var vecNorm = 1f
+    if (norm) {
```

Comment:
I don't mind much about make it always normalized. Just FYI: for current brute-force kNN implementation in findSynonyms, the unnormalized version does save potentially millions of division operation when the vocabulary is millions, of course it is still in at most seconds saved. When caller only care about top K without needs of metrics and call it for all their interested words, the minor save of time could be multiplied again.